scarb/src/compiler/compilers/lib.rs

```diff
@@ -127,3 +131,25 @@ impl Compiler for LibCompiler {
         Ok(())
     }
 }
+
+fn validate_compiler_config(
+    db: &RootDatabase,
+    compiler_config: &CompilerConfig<'_>,
+    ws: &Workspace<'_>,
+) {
+    // Generally, lib target compilation should be driven by a certain objective (e.g. cairo-run,
+    // test framework, etc.), expressed by the plugin set with executables definition.
+    // This does not apply to debug build (expressed by `replace_ids` flag),
+    // which is a goal by itself.
+    // See starkware-libs/cairo#5440 for more context.
+    let executable_plugin = db
+        .macro_plugins()
+        .iter()
+        .any(|plugin| !plugin.executable_attributes().is_empty());
+    if !executable_plugin && !compiler_config.replace_ids {
+        ws.config().ui().warn(
+            "no executable plugin found, and `replace_ids` is disabled, \
```

Comment:
As a Cairo code developer who has no knowledge of compiler/scarb internals (i.e. I only know I can write some code and test it with `snforge`), I would have no idea what this message is about 😕 

The term `executable plugin` is vague to the user. `replace_ids` also would probably be because by itself it is not stating that this is about compiler configuration flag.

This warning is also missing information how to fix it. It'd be best to include several notes that will suggest code snippets to put in Scarb.toml, or tell what commands user might want to actually run.